isis/tests/FunctionalTestsApolloFindRx.cpp

```diff
@@ -0,0 +1,93 @@
+#include "Fixtures.h"
+#include "Pvl.h"
+#include "PvlGroup.h"
+#include "TestUtilities.h"
+
+#include "apollofindrx.h"
+
+#include "gtest/gtest.h"
+
+using namespace Isis;
+
+static QString APP_XML = FileName("$ISISROOT/bin/xml/apollofindrx.xml").expanded();
+
+TEST_F(SmallCube, FunctionalTestApolloFindRxDefault) {
+  PvlGroup reseaus("Reseaus");
+  PvlKeyword samples = PvlKeyword("Sample", "2");
+  samples += "4"; 
+  samples += "6"; 
+  samples += "8";
+
+  PvlKeyword lines = PvlKeyword("Line", "2");
+  lines += "4"; 
+  lines += "6"; 
+  lines += "8";
+
+  PvlKeyword types = PvlKeyword("Type", "5");
+  types += "5"; 
+  types += "5"; 
+  types += "5";
+
+  PvlKeyword valid = PvlKeyword("Valid", "1");
+  valid += "1"; 
+  valid += "1"; 
+  valid += "1";
+
+  reseaus += lines; 
+  reseaus += samples; 
+  reseaus += types;
+  reseaus += valid; 
+  reseaus += PvlKeyword("Status", "Nominal");
+  
+  std::istringstream instStr (R"(
+    Group = Instrument
+        SpacecraftName = "APOLLO 15"
+        InstrumentId   = METRIC
+        TargetName     = MOON
+        StartTime      = 1971-08-01T14:58:03.78
+    End_Group
+  )");
+  
+  PvlGroup instGroup; 
+  instStr >> instGroup; 
+  
+  Pvl *lab = testCube->label();
+  lab->findObject("IsisCube").addGroup(reseaus);
+  lab->findObject("IsisCube").addGroup(instGroup);
+
+  QTemporaryDir prefix;
+  QVector<QString> args = {"tolerance=0.125"};
+
+  UserInterface options(APP_XML, args);
+  try {
+    apollofindrx(testCube, options);
+  }
+  catch (IException &e) {
+    FAIL() << "Call failed, Unable to process cube: " << e.what() << std::endl;
+  }
+
+  Pvl newLab = *testCube->label();
+
+  PvlGroup newReseaus = newLab.findObject("IsisCube").findGroup("Reseaus");
+  PvlKeyword testKeyword = newReseaus.findKeyword("Line");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, testKeyword[0], "-98.0");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, testKeyword[1], "-205.0");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, testKeyword[2], "-212.0");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, testKeyword[3], "-446.0");
+   
+  testKeyword = newReseaus.findKeyword("Sample");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, testKeyword[0], "-98.0");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, testKeyword[1], "-205.0");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, testKeyword[2], "-321.0");
+  EXPECT_PRED_FORMAT2(AssertQStringsEqual, testKeyword[3], "-446.0");
```

Comment:
If this do go in as-is, does it make sense to open a testing issue about making this test higher quality? I think those issues are beneficial, especially when we are on boarding new devs.

Reply:
The original test was not just a did it run test. It operates in place and updates the label with if the marks are valid or not and their refined locations if they are. If you compare the input label of the cube with the output label in the truthdata there are significant differences, which get checked in the test.

Reply:
The left is the input label and the right is the output label. You can see the files at `/work/users/jmapel/ISIS3/isis/src/apollo/apps/apollofindrx/tsts/default`

![Screen Shot 2020-10-14 at 10 00 58 AM](https://user-images.githubusercontent.com/22942817/96021395-34988180-0e04-11eb-9df7-4544181a7c52.png)

Reply:
I'm not questioning whether or not the data changes, what I am saying is that the test seems to mostly test an arbitrary inout/output state. Most of it's value is in checking whether or not the application state changed which seems to be accomplished by having a color gradient just as well as having an actual image. The test can be made better by rigorously checking accuracy, but considering the time it would require the justification is not there for this sprint. 

I'll make an issue to update this to be more rigorous but I personally am okay with getting the data out of ISIS and merging assuming no other issues.